### PR TITLE
setWaterLevel: add bIncludeWorldSeaLevel and bIncludeOutsideWorldLevel

### PR DESCRIPTION
**Pull request by TheNormalnij**
This PR makes it possible to change the water level outside the game world.

New syntax:
```lua
bool setWaterLevel ( float level [, bool includeWaterFeatures = true, bool includeWaterElements = true, bool includeWorldSea = true, bool includeOutsideWorld = false ] )
```
Reopened #1342 #889

### CODE REVIEW

File: Client/game_sa/CWaterManagerSA.cpp [hunk 1]
```diff
@@ -742,17 +742,25 @@ bool CWaterManagerSA::SetPositionWaterLevel(const CVector& vecPosition, float fL
...
-        if (bIncludeWorldNonSeaLevel || !m_Vertices[i].IsWorldNonSeaLevel())
-            vecVertexPos.fZ = fLevel;
-        m_Vertices[i].SetPosition(vecVertexPos, pChangeSource);
+        for (DWORD i = 0; i < NUM_DefWaterVertices; i++)
+        {
+            m_Vertices[i].GetPosition(vecVertexPos);
+            if ((bIncludeWorldNonSeaLevel && m_Vertices[i].IsWorldNonSeaLevel()) || (bIncludeWorldSeaLevel && !m_Vertices[i].IsWorldNonSeaLevel())) 
+                vecVertexPos.fZ = fLevel;
+            m_Vertices[i].SetPosition(vecVertexPos, pChangeSource);
+        }
```
Comment (Pirulax) [inline Client/game_sa/CWaterManagerSA.cpp:758]: Not sure what's the type of `m_Vertices`, but Im pretty sure you could use a for based range loop (eg.: `for(auto& vertex : m_Vertices)`

**Comment by Pirulax** [comment]
Seems okay to me.

File: Client/game_sa/CWaterManagerSA.cpp (hunk 1 above)
Comment (TheNormalnij) [reply Client/game_sa/CWaterManagerSA.cpp:758]: NUM_DefWaterVertices is constant value, I think that i will leave this.

File: Client/game_sa/CWaterManagerSA.cpp [hunk 2]
```diff
@@ -768,6 +776,17 @@ bool CWaterManagerSA::SetPolyWaterLevel(CWaterPoly* pPoly, float fLevel, void* p
     return true;
 }
 
+void CWaterManagerSA::SetOutsideWorldWaterLevel(float fLevel)
+{
+    // Outside world vertexes
```
Comment (StrixG) [inline Client/game_sa/CWaterManagerSA.cpp]:
```suggestion
    // Outside world vertices
```

**Review by StrixG — APPROVED** [review]

**Pull request merged**